launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -341,7 +341,7 @@ private String findAssembly(String scalaVersion) {
       libdir = new File(sparkHome, String.format("assembly/target/scala-%s", scalaVersion));
     }
 
-    final Pattern re = Pattern.compile("spark-assembly.*hadoop.*\\.jar");
+    final Pattern re = Pattern.compile("spark-assembly.*[hadoop]*.*\\.jar");
```

Comment:
Shouldn't this bit be `.*(hadoop)?.*` but then can't it just collapse to matching anything with `.*`?